File: hw14MongoDBAndGUI/Python/SchoolSystem/controller/student_controller.py
from controller.database import Database

class StudentController:

    def __init__(self):
        self.database = Database()

    def save_student(self, student):

        self.database.students.insert_one({
            "id": student.student_id,
            "firstName": student.first_name,
            "lastName": student.last_name,
            "age": student.age
        })

    def search_student(self, student_id):
        query = self.database.students.find_one({"id": student_id})
        return query

    def update_student(self, student):

        self.database.students.update_one(
            {"id": student.student_id},
            {
                "$set": {
                    "firstName": student.first_name,
                    "lastName": student.last_name,
                    "age": student.age
                }
            }
        )

    def delete_student(self, student_id):

        self.database.students.delete_one({"id": student_id})

File: hw14MongoDBAndGUI/JavaScript/SchoolSystem/controllers/studentController.js
const mongo = require("../database/mongo");

class StudentController {

    async saveStudent(student) {

        const db = await mongo.getDb();

        await db
            .collection("students")
            .insertOne(student);

        console.log("Student Saved");
    }

    async searchStudent(id) {

        const db = await mongo.getDb();

        return await db
            .collection("students")
            .findOne({ id });
    }

    async updateStudent(student) {

        const db = await mongo.getDb();

        await db
            .collection("students")
            .updateOne(
                { id: student.id },
                {
                    $set: {
                        firstName: student.firstName,
                        lastName: student.lastName,
                        age: student.age
                    }
                }
            );
    }

    async deleteStudent(id) {

        const db = await mongo.getDb();

        await db
            .collection("students")
            .deleteOne({ id });
    }
}

module.exports = StudentController;